File: core/src/main/java/com/eloja/core/entity/Juridica.java
package com.eloja.core.entity;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Data
public class Juridica extends Pessoa{

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Integer id;

    @Column(nullable = false)
    private String razaoSocial;

    @Column(nullable = false)
    private String nomeFantasia;

    @Column(nullable = false)
    private String cnpj;

    private String inscricao_estadual;

}

File: frontend-react/src/models/Juridica.ts
import Endereco from './Endereco';
import Telefone from './Telefone';

class Juridica {
  id?: number;
  razaoSocial: string;
  nomeFantasia: string;
  cnpj: string;
  inscricao_estadual?: string;
  endereco: Endereco;
  telefones: Telefone[];

  constructor(
    razaoSocial: string,
    nomeFantasia: string,
    cnpj: string,
    inscricao_estadual: string,
    endereco: Endereco,
    telefones: Telefone[],
    id?: number,
  ) {
    this.id = id;
    this.razaoSocial = razaoSocial;
    this.nomeFantasia = nomeFantasia;
    this.cnpj = cnpj;
    this.inscricao_estadual = inscricao_estadual;
    this.endereco = endereco;
    this.telefones = telefones;
  }
}

export default Juridica;
